src/chart/generateCategoricalChart.tsx

```diff
@@ -802,6 +803,7 @@ export interface CategoricalChartProps {
   data?: any[];
   layout?: LayoutType;
   stackOffset?: StackOffsetType;
+  stepRatioControl?: StepRatioControl;
```

Comment:
one idea I had was, instead of making this a separate parameter that gets passed to every chart at the highest level why don't we associate it with the domain (since thats where it is used). For example, we can keep everything as-is but allow an object as part of domain specification.

`domain={'auto'}`

OR

`domain={[0, 10000]}`

OR

`domain={(min, max) => [min, max]}

OR

`domain={{ domain: 'auto' , settings: stepRatioControl: 0.1 }}` 
(or maybe somehow only allow it if domain is auto so there is no confusion)


Or if not the above then

`domainSettings={{ stepRatioControl: 0.1 }}`

on the X/YAxis?

This way there is less ambiguity against what the parameter affects/changes. Otherwise it seems arbitrary and hard to follow like the discussion in the other thread...

Reply:
yeah I think combining this parameter with something is a good idea, but my concern is that it would require a lot of changes to the code that currently assumes the input value is an array containing two numeric values when the domain is set to `auto` - changing it to an object would add a lot messy. Another concern is that making it an object allows users to set any number, which might be problematic if the value is being set to too high/ low, without the enum type they would need an even better understanding of how to change the value. Anyway I like the idea, just wondering if there's a better way to combine this parameter